File: Back/Test/Models/Invoice.cs
using System;

namespace Test.Models
{
    public class Invoice
    {
        public DateTime LastEditionTime { get; set; } = DateTime.Now;
        public string Id { get; set; }
        public InvoiceStatus Status { get; set; } = InvoiceStatus.New;
        public decimal Price { get; set; }
        public PaymentMethod PaymentMethod { get; set; }

        public void Update(Invoice newInvoince)
        {
            LastEditionTime = DateTime.Now;
            Status = newInvoince.Status;
            Price = newInvoince.Price;
            PaymentMethod = newInvoince.PaymentMethod;
        }
    }
}

File: Front/src/app/invoice/models/invoice.ts
import { InvoiceStatus } from './invoiceStatus';
import { PaymentMethod } from './paymentMethod';

export class Invoice {
  constructor(public lastEditionTime?: string,
              public id?: string,
              public status: InvoiceStatus = InvoiceStatus.New,
              public price?: number,
              public paymentMethod?: PaymentMethod) {
  }

  isEqual(oldInvoice: Invoice) {
    return this.status === oldInvoice.status &&
      this.price == oldInvoice.price &&
      this.paymentMethod == oldInvoice.paymentMethod &&
      this.id === oldInvoice.id;
  }
}
